src/Compilers/Core/Portable/ConstantValue.cs

```diff
@@ -1,5 +1,7 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+#nullable enable
```

Comment:
💡 Should be `IEquatable<ConstantValue?>`

Reply:
The argument to `IEquatable<T>.Equals()` is `[AllowNull]T other`.

---
In reply to: [356883812](https://github.com/dotnet/roslyn/pull/40268#discussion_r356883812) [](ancestors = 356883812)

Reply:
Thus far we've been preferring `IEquatable<TheType?>`. Going to go with that pattern for now but we should discuss which is the more preferred approach. 

---
In reply to: [356891911](https://github.com/dotnet/roslyn/pull/40268#discussion_r356891911) [](ancestors = 356891911,356883812)